streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -125,7 +124,7 @@ private boolean productionExceptionIsFatal(final Exception exception) {
     ) {
         String errorLogMessage = LOG_MESSAGE;
         String errorMessage = EXCEPTION_MESSAGE;
-        if (exception instanceof RetriableException) {
+        if (exception instanceof TimeoutException) {
```

Comment:
@mjsax Do you know why we add the hint for any `RetriableException` instead of `TimeoutException`?

Reply:
`TimeoutException` extends `RetriableException`. Thus, I think catching `RetriableException` is correct.

Reply:
Maybe it's better to have two checks: first test for `TimeoutException` and if it's a different one, test for `RetriableException` second.

Reply:
It's not correct since the reason may not be timeout related, right? So, we may give the wrong hint.

Reply:
What would be the hint for `RetriableException`?

Reply:
> What would be the hint for `RetriableException`?

The current hint seem to be appropriate.

Reply:
It isn't. The hint is only appropriate if it's a TimeoutException. If it's any other RetriableException, it's not the right fix.

Reply:
Can you elaborate. Why is
```
You can increase the producer configs `retries` and `retry.backoff.ms` to avoid this error.
```
for suitable for a `RetriableException`? (With "current" I referred to the old/existing one, not the updated/new one.)

As said above, I think it would be best to have two hints -- one for timeout, and one for all other retriable cases.

Reply:
A couple of things:

- `retries` are infinite by default. We recommend that people don't set it unless they need to set it to 0 because they want no retries. In both of these cases, the hint to change `retries` is not helpful.
- `RetriableException`s are not meant to be thrown to users. They are an internal construct that we use. The fact that `TimeoutException` inherits from `RetriableException` is an implementation detail and one could say it was a mistake. You can search in the KafkaProducer javadoc and `RetriableExcepton` is not mentioned anywhere.

In any case, looking more at the code, there is no good mechanism to do what is being done here. So, we probably have to live with it.

Reply:
Thanks for the details. Seems I did not have full context.